connectors/sources/google_cloud_storage.py

```diff
@@ -199,8 +185,54 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def _validate_configurations(self):
+        """Validates whether user input is empty or not for configuration field.
+
+        Raises:
+            Exception: Configured keys can't be empty.
+        """
+        if self.configuration["service_account_credentials"] == "":
+            raise Exception("service_account_credentials can't be empty.")
+
+    def get_pem_format(self, private_key, max_split=-1):
+        """Convert key into PEM format.
+
+        Args:
+            private_key (str): Private_key in raw format.
+            max_split (int): Specifies how many splits to do. Defaults to -1.
+
+        Returns:
+            string: PEM format
+        """
+        private_key = private_key.replace(" ", "\n")
+        private_key = " ".join(private_key.split("\n", max_split))
+        private_key = " ".join(private_key.rsplit("\n", max_split))
+        return private_key
+
+    def _initialize_configurations(self):
+        """Initialize the ServiceAccountCreds"""
+        json_creds = json.loads(self.configuration["service_account_credentials"])
+        if (
+            not RUNNING_FTEST
```

Comment:
It feels a bit odd to have code specific to testing inside a connector. Can we solve this differently?

Reply:
It was done to prevent method calls explicitly in e2e tests only. I've removed the shield, either way for e2e, the if-block will not be executed.